Sources/GoodExtensions/Foundation/DateExtentions.swift

```diff
@@ -22,4 +22,44 @@ public extension GRActive where Base == Date {
         return Calendar.current.date(byAdding: component, value: number, to: base) ?? base
     }
 
+    /**
+    Subtracting returns a new date with the subtracted component
+     - parameter number: Number of component value to be subtracted
+     - parameter component: Component to add
+     - returns: New date with subracted component
+     */
+    func subtracting(number: Int, of component: Calendar.Component) -> Date {
+        return adding(number: -number, of: component)
+    }
+
+    /**
+     localizedPluralDateString returns a localized string
+     - parameter number: Number of component value to be added
+     - parameter component: Component to add
+     - parameter style: DateComponentsFormatter.UnitsStyle
+     - returns: Second part of the string with from localized date string
+     */
+    static func localizedPluralDateString(
+        from number: Int,
+        of component: Calendar.Component,
+        with style: DateComponentsFormatter.UnitsStyle,
+        calendar: Calendar = .sk_CET
+    ) -> String {
+        let formatter = DateComponentsFormatter()
+        formatter.unitsStyle = style
+        let fromDate = Date()
+        let toDate = fromDate.gr.adding(number: number, of: component)
+
+        let calendar = Calendar.current
+        let components = calendar.dateComponents(
+            [component],
+            from: fromDate,
+            to: toDate
+        )
+        if let substring = formatter.string(from: components)?.split(separator: " ")[safe: 1] {
```

Comment:
Nebude nahodou toto fungovat iba s konkretnym unitStylom? V pripade ze by som pouzit positional tak by to nefungovalo ci? Nechceme tam dat nejaky konkretny default, ktory sa neda menit, alebo ten separator priradit na zaklade typu?


BTW, aky bol konkretny scnar, kde ste toto pouzili?

Reply:
True, fungovalo to v testoch len s .full to mozem zafixovat.
Scenar bol taky ze som potreboval vytiahnut do 2 roznych elementov cislo a string co k tomu cislu nalezi lebo obe casti stringu boli samostatne labele s roznymi constraintami. Myslim ze ten design ale nieje nezvycajny a urcite by sa mohol vyskutnt v buducnosti
<img width="283" alt="Screenshot 2022-02-13 at 09 39 49" src="https://user-images.githubusercontent.com/78155066/153745590-27ef0b03-f862-4e8b-bfad-ff8708a83ede.png">

Reply:
Super, tak to je pekne riesenie, uplne chapem. Tak skus upravit bud fixne na .full alebo pre kazdy variant samostatne.